src/main/java/org/embulk/input/jira/Issue.java

```diff
@@ -0,0 +1,155 @@
+package org.embulk.input.jira;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonNull;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.StreamSupport;
+
+public class Issue
+{
+    private JsonObject flatten;
+    private JsonObject json;
+
+    public Issue(JsonObject original)
+    {
+        this.json = original;
+    }
+
+    public JsonElement fetchValue(String path)
+    {
+        List<String> keys = new ArrayList<>(Arrays.asList(path.split("\\.")));
+        return fetch(json, keys);
+    }
+
+    private JsonElement fetch(JsonElement json, List<String> keys)
+    {
+        if (json == null || json.isJsonNull()) {
+            return JsonNull.INSTANCE;
+        }
+        else if (keys.isEmpty() || (json.isJsonArray() && json.getAsJsonArray().size() == 0)) {
+            return json;
+        }
+        String key = keys.get(0);
+        keys.remove(0);
+        if (json.isJsonArray()) {
+            JsonArray arrays = new JsonArray();
+            StreamSupport.stream(json.getAsJsonArray().spliterator(), false)
+                        .forEach(obj -> {
+                            if (obj.isJsonObject()) {
+                                arrays.add(obj.getAsJsonObject().get(key));
+                            }
+                            else {
+                                arrays.add(obj);
+                            }
+                        });
+            return fetch(arrays, keys);
+        }
+        else {
+            return fetch(json.getAsJsonObject().get(key), keys);
+        }
+    }
+
+    public void toRecord()
+    {
+        flatten = new JsonObject();
+        manipulatingFlattenJson(json, "");
+    }
+
+    public JsonObject getFlatten()
+    {
+        return flatten;
+    }
+
+    private void manipulatingFlattenJson(JsonElement in, String prefix)
+    {
+        if (in.isJsonObject()) {
+            JsonObject obj = in.getAsJsonObject();
+            // NOTE: If you want to flatten JSON completely, please remove this if and addHeuristicValue
+            if (StringUtils.countMatches(prefix, ".") > 1) {
+                addHeuristicValue(obj, prefix);
+                return;
+            }
+            for (Entry<String, JsonElement> entry : obj.entrySet()) {
+                String key = entry.getKey();
+                JsonElement value = entry.getValue();
+                manipulatingFlattenJson(value, appendPrefix(prefix, key));
+            }
+        }
+        else if (in.isJsonArray()) {
+            JsonArray arrayObj = in.getAsJsonArray();
+            boolean isAllJsonObject = arrayObj.size() > 0 ? StreamSupport.stream(arrayObj.spliterator(), false).allMatch(x -> x.isJsonObject()) : false;
```

Comment:
Done. 
The reason I do not use parallel because the size normally small.
Likely it is < 10. Sometimes only 1 or 2.
Hence using parallel is a waste. I suppose.